docs/data/sdks/android-kotlin/migration.md

```diff
@@ -1,13 +1,247 @@
+    -   eventProperties.put("buttonColor", "primary")
+    - } catch (e: JSONException) {
+    -   System.err.println("Invalid JSON")
+    -   e.printStackTrace()
+    - }
+    - client.logEvent("Button Clicked", eventProperties)
+
+    + client.track(
+    +     "Button Clicked",
+    +     mutableMapOf<String, Any?>("buttonColor" to "primary")
+    + )
+    ```
+
+=== "Java"
+
+    ```diff
+    - import org.json.JSONException;
+    - import org.json.JSONObject;
+
+    - JSONObject eventProperties = new JSONObject();
+    - try {
+    -   eventProperties.put("buttoncolor", "primary");
+    - } catch (JSONException e) {
+    -   System.err.println("Invalid JSON");
+    -   e.printStackTrace();
+    - }
+    - client.logEvent("Button Clicked", eventProperties);
+
+    + client.track("Button Clicked", new HashMap() {{
+    +     put("buttoncolor", "primary");
+    + }});
+    ```
+
+### Flush events
+
+Unset events are stored in a buffer and flushed (sent) on app close by default. Events are flushed based on which criteria is met first: `flushIntervalMillis` or `flushQueueSize`.
+
+You can disable flushing by setting `flushEventsOnClose` to `false`.
+
+#### `flush()`
+
+You can also force the SDK to upload unsent events. The `uploadEvents()` API maps to `flush()`.
+
+=== "Kotlin"
+
+    ```diff
+    - client.uploadEvents()
+    + client.flush()
+    ```
+
+=== "Java"
+
+    ```diff
+    - client.uploadEvents();
+    + client.flush();
+    ```
+
+### Set user properties
+
+#### `identify()`
+
+The `identify()` API remains the same
+
+=== "Kotlin"
+
+    ```diff
+    val identify = Identify()
+    identify.set("location", "LAX")
+    client.identify(identify)
+    ```
+
+=== "Java"
+
+    ```diff
+    Identify identify = new Identify();
+    identify.set("location", "LAX");
+    client.identify(identify);
+    ```
+
+### Set group properties
+
+### `groupIdentify()`
+
+The `groupIdentify()` API remains the same.
+
+=== "Kotlin"
+
+    ```diff
+    val groupType = "plan"
+    val groupName = "enterprise"
+
+    val identify = Identify().set("key", "value")
+    client.groupIdentify(groupType, groupName, identify)
+    ```
+
+=== "Java"
+
+    ```diff
+    String groupType = "plan";
+    Object groupName = "enterprise";
+
+    Identify identify = new Identify().set("key", "value");
+    client.groupIdentify(groupType, groupName, identify);
+    ```
+
+### Track revenue
+
+#### `logRevenueV2()`
+
+The `logRevenueV2()` API maps to `revenue()`.
+
+=== "Kotlin"
+    ```diff
+    val revenue = Revenue()
+    revenue.productId = "com.company.productId"
+    revenue.price = 3
+    revenue.quantity = 2
+    - client.logRevenueV2(revenue)
+    + client.revenue(revenue)
+    ```
+
+=== "Java"
+
+    ```diff
+    Revenue revenue = new Revenue()
+    revenue.setProductId("com.company.productId");
+    revenue.setPrice(3);
+    revenue.setQuantity(2);
+    - client.logRevenueV2(revenue);
+    + client.revenue(revenue);
+    ```
+
+#### Revenue verification
```

Comment:
Thanks ❤️

Reply:
No problem. Thanks for confirming with data pipeline team.